### PR TITLE
Add GitHub action to trigger versioned builds in docs repository

### LINKED ISSUE #7211: Create, preserve, and host a version of the docs for each Prefect release
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

A user on Prefect on any version other than the latest version of Prefect doesn't have access to the docs corresponding to their Prefect version. So a user on 2.0 can't make deployments from a Python file, while the docs show them that they can.

### Describe the proposed behavior

With users of different versions of Prefect, @madkinsz and I were discussing how nice it would be for users to be able to refer to specific doc versions. He indicated it wouldn't be too tricky to make the historic doc versions. 

Read the Docs makes it easy to host doc versions that correspond to each release. Not sure what other options are available. 

### Example Use

A user on 2.x could see what the docs look like for their Prefect version.

### Additional context

Curious to hear your thoughts @tpdorsey.

### PR DESCRIPTION
**Pull request by rpeden**
This PR adds a GitHub action to trigger builds in the versioned docs repository. 

The workflow has three triggers:
- A new release is published
- A change to the `docs` directory is merged to `main`
- A change to `netlify.toml` is merged to `main`

#### New release trigger
Runs of this workflow  triggered by a new release being published start the build + publish working in the `docs` repo for the newly-published release version.

#### Other triggers
Both other triggers will kick off the `docs` workflow for the 'unreleased' version of the docs. This `unreleased` version appears at the bottom of the version selector and can be used to view the latest changes to the docs that aren't yet included in a Prefect release. The README in the `docs` repository contains instructions on how to hide this version from the selector if desired.

Closes #7211 

### QA Done
- I ran the `curl` command in this action using my own PAT using `unreleased` and several other release versions to ensure that the repository dispatch action triggers auto-deployed docs builds as expected.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.

### CODE REVIEW

File: .github/workflows/build-docs.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    types:
+      - published
+
+jobs:
+  update_docs:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Trigger docs update in PrefectHQ/docs
+      env:
+        REPO_ACCESS_TOKEN: ${{ secrets.PREFECT_DOCS_REPO_ACCESS_TOKEN }}
```
Comment (jamiezieziula) [inline .github/workflows/build-docs.yaml:20]:
```suggestion
        REPO_ACCESS_TOKEN: ${{ secrets.DOCS_CONTENTS_RW }}
```

**Comment by rpeden** [comment]
@madkinsz @serinamarie sorry to bug you about this one, but I can't update it without forking the repo and making a new PR (and I can do that if you'd prefer). I think this just needs one change before it's ready:

```
"Authorization: Bearer $PREFECT_DOCS_REPO_ACCESS_TOKEN"
```

should now be:
```
Authorization: Bearer $REPO_ACCESS_TOKEN
```

Then, it should work. Last week, I tested the repo dispatch call by running the Curl command from this GH action with my own PAT and the docs repo built and deployed everything as expected.

**Comment by serinamarie** [comment]
> "Authorization: Bearer $REPO_ACCESS_TOKEN"

Hey! Yeah I can update it to see if it works :)

**Comment by rpeden** [comment]
Thank you very much, @serinamarie! I'll move this PR out of draft status. If it gets merged before the next release, the new docs version should auto-build and deploy as soon as the release is published.

**Review by serinamarie — APPROVED** [review]
Looks good to me!

**Comment by jawnsy** [comment]
Thanks @rpeden! Merging now

**Comment by rpeden** [comment]
ohyeah

**Pull request merged**